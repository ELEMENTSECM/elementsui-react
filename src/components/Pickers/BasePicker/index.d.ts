import * as React from 'react';

export interface BasePickerProps {
	/**
	 * Sets focus to the focus zone
	 */
	focus: boolean;
	/**
	 * Set focus to the input
	 */
	focusInput: boolean;
	/**
	 * Callback when filtering
	 */
	onResolveSuggestions: (...args: any[]) => any;
	/**
	 * Text input suggestion
	 */
	getTextFromItem?: (...args: any[]) => any;
	/**
	 * Suggestion dropdown props
	 */
	pickerSuggestionsProps?: Object;
	/**
	 * Input props
	 */
	inputProps?: Object;
	/**
	 * Restrict the amount of selectable items.
	 */
	itemLimit?: number;
	/**
	 * User-defined styling
	 */
	styles?: (...args: any[]) => any;
}

declare const BasePicker: React.SFC<BasePickerProps>;

export default BasePicker;
