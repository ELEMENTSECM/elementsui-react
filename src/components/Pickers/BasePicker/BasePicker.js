import * as React from 'react';
import PropTypes from 'prop-types';
import { TagPicker } from 'office-ui-fabric-react/lib/components/pickers/TagPicker/TagPicker';
import { styles } from './BasePicker.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** BasePicker example */
export function BasePicker(props) {
	const {
		focus,
		focusInput,
		onResolveSuggestions,
		getTextFromItem,
		pickerSuggestionsProps,
		inputProps,
		className,
		theme,
		styles
	} = props;
	const classNames = classNamesFunction()(styles, props);
	return (
		<TagPicker
			className={classNames.root}
			focus={focus}
			focusInput={focusInput}
			onResolveSuggestions={onResolveSuggestions}
			getTextFromItem={getTextFromItem}
			pickerSuggestionsProps={pickerSuggestionsProps}
			inputProps={inputProps}
		/>
	);
}

BasePicker.propTypes = {
	/** Sets focus to the focus zone */
	focus: PropTypes.bool.isRequired,
	/** Set focus to the input  */
	focusInput: PropTypes.bool.isRequired,
	/** Callback when filtering*/
	onResolveSuggestions: PropTypes.func.isRequired,
	/** Text input suggestion */
	getTextFromItem: PropTypes.func,
	/** Suggestion dropdown props */
	pickerSuggestionsProps: PropTypes.object,
	/** Input props */
	inputProps: PropTypes.object
};

export default styled(customizable('BasePicker', ['theme'])(BasePicker), styles);
