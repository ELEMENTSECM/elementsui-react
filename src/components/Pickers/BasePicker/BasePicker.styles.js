export const getStyles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-BasePicker',
			{
				// place your styles here
			},
			className
		]
	};
};
