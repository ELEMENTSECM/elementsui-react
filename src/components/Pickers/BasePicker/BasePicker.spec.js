import React from 'react';
import renderer from 'react-test-renderer';
import BasePicker from './BasePicker';

const items = () => {
	return ['Andreas', 'Eren', 'Haakon', 'Jan Rune', 'Jarle'];
};

describe('BasePickerProps', () => {
	test('displays basepicker correctly', () => {
		const tree = renderer
			.create(
				<BasePicker
					htmlId="basePicker"
					onResolveSuggestions={items}
					focus={false}
					focusInput={true}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
