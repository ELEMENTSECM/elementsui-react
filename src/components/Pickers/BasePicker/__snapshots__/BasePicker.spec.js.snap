// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePickerProps displays basepicker correctly 1`] = `
<div
  className="ms-BasePicker ms-BasePicker root-40"
  onKeyDown={[Function]}
>
  <div
    aria-describedby={undefined}
    aria-labelledby={undefined}
    className="ms-FocusZone"
    data-focuszone-id="FocusZone1"
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDownCapture={[Function]}
    role="presentation"
  >
    <div
      className="ms-SelectionZone"
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
      onFocusCapture={[Function]}
      onKeyDown={[Function]}
      onKeyDownCapture={[Function]}
      onMouseDown={[Function]}
      onMouseDownCapture={[Function]}
      role="presentation"
    >
      <div
        className="ms-BasePicker-text pickerText_4fe0caaf"
        role="list"
      >
        <input
          aria-activedescendant={undefined}
          aria-controls="selected-suggestion-alert"
          aria-expanded={false}
          aria-haspopup="true"
          aria-owns={undefined}
          autoCapitalize="off"
          autoComplete="off"
          className="ms-BasePicker-input pickerInput_4fe0caaf"
          data-lpignore={true}
          disabled={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          role="combobox"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
